lib/spack/spack/filesystem_view.py

```diff
@@ -729,6 +734,45 @@ def _check_no_ext_conflicts(self, spec):
                      'Skipping already activated package: %s' % spec.name)
 
 
+class SimpleFilesystemView(FilesystemView):
+    """File system view without the option to revert and without extension business."""
+
+    def __init__(self, root, layout, **kwargs):
+        super(SimpleFilesystemView, self).__init__(root, layout, **kwargs)
+
+    def add_specs(self, *specs, **kwargs):
+        assert all((s.concrete for s in specs))
+        if len(specs) == 0:
+            return
+
+        if kwargs.get("exclude", None):
+            specs = set(filter_exclude(specs, kwargs["exclude"]))
+
+        ignore = lambda path: os.path.basename(path) == '.spack'
+        visitor = MergeVisitor(ignore=ignore)
+
+        # todo, group by destination?
+        # assumes single dst/ now.
```

Comment:
I think this is as simple as

1. create dict from dst to specs using `self.get_projection_for_spec(spec)`
2. do the rest of the method for each dst

Do you see any problems with this part of increasing functionality?

Reply:
It's fixed now